tactics/hints.ml

```diff
@@ -66,6 +66,9 @@ let decompose_app_bound t =
     | Proj (p, c) -> ConstRef (Projection.constant p), Array.cons c args
     | _ -> raise Bound
 
+let secvars_of_hyps hyps =
+  List.fold_left (fun l id -> Id.Pred.add id l) Id.Pred.empty hyps
+
```

Comment:
Here I can actually be more efficient by making a predicate starting from the section context and not testing each of the hyps variables. Will be in my new patch.